### ISSUE TITLE
Wrong unresolved error on defmacro

### ISSUE THREAD

**Issue by licht1stein** [clj-kondo/clj-kondo#2291]
[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! ]

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**
clj-kondo v2024.02.12

**macro usage**
The bug is related to a macro, but it was linting fine before. Macro config:

```clojure
{:lint-as {b.jobs.helpers/defjob clojure.core/defn}}
```

**platform**
Native, installed via homebrew

**editor**
Emacs

**problem**
After defining a macro, it's invocation is not recognised:

```sh
./components/jobs/src/b/jobs/core.clj:8:34: warning: Unresolved var: defjob
./components/jobs/src/b/jobs/helpers.clj:51:4: error: Unresolved symbol: defjob
./components/jobs/src/b/jobs/helpers.clj:51:11: error: Unresolved symbol: foo
```

**repro**
```clojure
(ns b.jobs.helpers)

(def job-registry (atom #{}))

(defn register-job
  [{:keys [:job/id] :as job}]
  (when (empty? (filter #(= id (:job/id %)) @job-registry))
    (swap! job-registry conj job)))

(defn response
  [{:keys [id ok? message result]}]
  (let [resp (cond->
              {:id id
               :ok? (if (false? ok?) ok? true)}
               (some? message) (assoc :message message)
               (some? result) (assoc :result result))]
    resp))

(defmacro defjob
  "Like defn, but turns a function into a job. Docstring required."
  [name docstring args body]
  (let [job-id (keyword name)]
    `(do
       (def ~name
         ~docstring
         (fn ~args
           (try
             (response {:result ~body
                        :id ~job-id})
             (catch Exception e#
               (response {:ok? false
                          :id ~job-id
                          :result (str e#)})))))

       (register-job {:job/id ~job-id
                      :job/description ~docstring
                      :job/callback ~name})
       ~name)))

;; This usage is marked as error
(defjob foo "bar" [] 1)
```

**config**
I didn't modify config since upgrading clj-kondo.

**expected behavior**
Do not mark `defjob` usage as error.

**Comment by borkdude** [comment]
I can't reproduce this.

```
borkdude@m1 ~/dev/scratch $ clj-kondo --lint src/repro.clj
linting took 52ms, errors: 0, warnings: 0
borkdude@m1 ~/dev/scratch $ cat .clj-kondo/config.edn
{:lint-as {
           repro/defjob clojure.core/defn}}
```

Can you perhaps make a separate Github repository and show a stepwise reproduction? 
Perhaps there is something stateful going on in your project. Perhaps you have multiple files with the same namespace.

**Comment by goshatch** [comment]
I might be running into the same thing: I'm using [clj-simple-router](https://github.com/tonsky/clj-simple-router), which provides [a `routes` macro](https://github.com/tonsky/clj-simple-router/blob/main/src/clj_simple_router/core.clj#L144). I'm using it [in this source file](https://github.com/goshatch/cljwiki/blob/main/src/wiki/core.clj#L43).

When I open the file in Emacs with `clj-kondo`, I'm getting false positive "Unresolved symbol" errors:

![telegram-cloud-photo-size-4-5823651645794402540-y](https://github.com/user-attachments/assets/3b283c13-1306-4a93-b5c2-ece151818530)

However, the code works as expected.

**Comment by borkdude** [comment]
@goshatch clj-kondo uses static analysis and doesn't run your program. As such i doesn't recognize user-defined macros without a little help. You can check here for the options that clj-kondo has to deal with this: https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md#unrecognized-macros

**Comment by borkdude** [comment]
As the original issue wasn't reproducible by me, I'm closing this, unless there is a follow up.
